signer/rpc_and_client_test.go

```diff
@@ -80,73 +81,124 @@ func (s stubServer) Sign(ctx context.Context, sr *pb.SignatureRequest) (*pb.Sign
 	require.Error(t, err)
-	require.True(t, strings.Contains(err.Error(), "Timed out"))
+	require.Contains(t, err.Error(), context.DeadlineExceeded.Error())
 }
 
 // CheckHealth succeeds if KM is healthy and reachable.
 func TestHealthCheckKMHealthy(t *testing.T) {
-	s := getStubbedHealthServer(func() (map[string]string, error) {
-		return nil, nil
-	})
+	hs := health.NewServer()
+	hs.SetServingStatus(notary.HealthCheckKeyManagement, healthpb.HealthCheckResponse_SERVING)
+
+	s := getStubbedHealthServer(hs)
 	signerClient, _, cleanup := setUpSignerClient(t, s)
 	defer cleanup()
-	require.NoError(t, signerClient.CheckHealth(1*time.Second))
+	require.NoError(t, signerClient.CheckHealth(1*time.Second, notary.HealthCheckKeyManagement))
 }
 
 // CheckHealth fails immediately if not connected to the server.
 func TestHealthCheckConnectionDied(t *testing.T) {
-	s := getStubbedHealthServer(func() (map[string]string, error) {
-		return nil, nil
-	})
+	hs := health.NewServer()
+	hs.SetServingStatus(notary.HealthCheckKeyManagement, healthpb.HealthCheckResponse_NOT_SERVING)
+
+	s := getStubbedHealthServer(hs)
 	signerClient, conn, cleanup := setUpSignerClient(t, s)
 	defer cleanup()
 	conn.Close()
-	require.Error(t, signerClient.CheckHealth(1*time.Second))
+	require.Error(t, signerClient.CheckHealth(1*time.Second, notary.HealthCheckKeyManagement))
+}
+
+// TestHealthCheckForOverallStatus query for signer's overall health status
+// with the empty provided service name.
+func TestHealthCheckForOverallStatus(t *testing.T) {
+	hs := health.NewServer()
+
+	s := getStubbedHealthServer(hs)
+	signerClient, _, cleanup := setUpSignerClient(t, s)
+	defer cleanup()
+
+	// both of the service are NOT SERVING, expect the health check for overall status to be failed.
+	hs.SetServingStatus(notary.HealthCheckKeyManagement, healthpb.HealthCheckResponse_NOT_SERVING)
+	hs.SetServingStatus("grpc.health.v1.Health.Signer", healthpb.HealthCheckResponse_NOT_SERVING)
+	err := signerClient.CheckHealth(1*time.Second, notary.HealthCheckOverall)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "NOT_SERVING, want SERVING")
+
+	// change the status of KeyManagement to SERVING and keep the status of Siger
+	// still be NOT SERVING, expect the health check for overall status to be failed.
+	hs.SetServingStatus(notary.HealthCheckKeyManagement, healthpb.HealthCheckResponse_SERVING)
+	err = signerClient.CheckHealth(1*time.Second, notary.HealthCheckOverall)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "NOT_SERVING, want SERVING")
+
+	// change the status of Signer to SERVING, expect the health check for overall status to success.
+	hs.SetServingStatus("grpc.health.v1.Health.Signer", healthpb.HealthCheckResponse_SERVING)
+	err = signerClient.CheckHealth(1*time.Second, notary.HealthCheckOverall)
+	require.NoError(t, err)
+
+}
+
+// TestHealthCheckUnExistsService query for an un-exists service's health status
```

Comment:
nit: can we change `an un-exists` to `a nonexistent`, both here, in the function name, and in the comment below?